### PR TITLE
Update VBox Guest Additions to latest version

### PR DESCRIPTION
**Pull request by felipecassiors**
To do this, we switch the installation from the package manager to
directly from the ISO.

Signed-off-by: Felipe de Cássio Rocha Santos <felipesantos@inatel.br>

### CODE REVIEW

**Comment by felipecassiors** [comment]
This is because of this:

 ``` log
==> default: Checking for guest additions in VM...
    default: The guest additions on this VM do not match the installed version of
    default: VirtualBox! In most cases this is fine, but in rare cases it can
    default: prevent things such as shared folders from working properly. If you see
    default: shared folder errors, please make sure the guest additions within the
    default: virtual machine match the version of VirtualBox you have installed on
    default: your host and reload your VM.
    default:
    default: Guest Additions Version: 5.2.32
    default: VirtualBox Version: 6.0
```

**Comment by ruzickap** [comment]
May I ask you which images are giving you the error you described above?

**Comment by felipecassiors** [comment]
The latest version of Ubuntu 18.04 desktop:
https://app.vagrantup.com/peru/boxes/ubuntu-18.04-desktop-amd64/versions/20190901.01

However, I tried to install the VirtualBox Guest Additions trough `apt-get` on my own VM and it is indeed outdated.

**Comment by ruzickap** [comment]
I agree... 
VirtualBox Guest Additions are outdated on "old" Ubuntu, but they should be fully supported. The best is using the "latest" one like you suggest (but they are only supported by Oracle).

Actually I compared the size of the installed packages by your changed scripts and the original one, which only install 2 packages (virtualbox-guest-x11, virtualbox-guest-utils) with relatively small size.

Here are the examples of the "original" images and new ones build using your change:

```bash
pruzicka@ubuntu:/var/tmp$ ls -l --block-size=M packer-templates-images-default/*.box packer-templates-images-new-virtualbox-guest/*.box
-rw-rw-r-- 1 pruzicka pruzicka  440M Dec 16 22:11 packer-templates-images-default/ubuntu-14.04-server-amd64-virtualbox.box
-rw-rw-r-- 1 pruzicka pruzicka  495M Dec 16 21:58 packer-templates-images-default/ubuntu-16.04-server-amd64-virtualbox.box
-rw-rw-r-- 1 pruzicka pruzicka  603M Dec 16 21:46 packer-templates-images-default/ubuntu-18.04-server-amd64-virtualbox.box
-rw-rw-r-- 1 pruzicka pruzicka 1618M Dec 16 21:06 packer-templates-images-default/ubuntu-19.04-desktop-amd64-virtualbox.box
-rw-rw-r-- 1 pruzicka pruzicka 1579M Dec 16 21:32 packer-templates-images-default/ubuntu-19.10-desktop-amd64-virtualbox.box

-rw-rw-r-- 1 pruzicka pruzicka  492M Dec 17 00:18 packer-templates-images-new-virtualbox-guest/ubuntu-14.04-server-amd64-virtualbox.box
-rw-rw-r-- 1 pruzicka pruzicka  568M Dec 17 00:03 packer-templates-images-new-virtualbox-guest/ubuntu-16.04-server-amd64-virtualbox.box
-rw-rw-r-- 1 pruzicka pruzicka  671M Dec 16 23:49 packer-templates-images-new-virtualbox-guest/ubuntu-18.04-server-amd64-virtualbox.box
-rw-rw-r-- 1 pruzicka pruzicka 1628M Dec 16 23:06 packer-templates-images-new-virtualbox-guest/ubuntu-19.04-desktop-amd64-virtualbox.box
-rw-rw-r-- 1 pruzicka pruzicka 1618M Dec 16 23:33 packer-templates-images-new-virtualbox-guest/ubuntu-19.10-desktop-amd64-virtualbox.box
```

In short images build by using your PR are about 60Mb bigger (603M vs 671M) which is about 10%.

Actually 10% is quite a lot.

Does the latest version of `virtualbox-guest-utils` and `virtualbox-guest-x11` bring some real value comparing to the "older" ones which are in official Ubuntu?

There should be a good reason for creating 10% bigger images.

Thank you...

**Comment by felipecassiors** [comment]
You're right: I never thought about the size difference. I just wondered because Vagrant seems to do not like the mismatch of the Additions version, so as the VirtualBox itself. Despite it, I have not seen any malfunction because of this. 

One interesting thing is that [hashicorp/bionic64](https://app.vagrantup.com/hashicorp/boxes/bionic64) seems to have the right version of the Additions, but sadly I did not find its source code.

**Comment by ruzickap** [comment]
It would be nice how the "hashicorp guys" are building their images.

Currently I prefer to keep the "VBox Guest Additions" versions provided officially by Ubuntu and keep the images smaller.

If there will be some "killer feature" in the future/latest versions of "VBox Guest Additions" then we can discuss it again...

Thank you for your help :-)

**Comment by felipecassiors** [comment]
> It would be nice how the "hashicorp guys" are building their images.
> 
> Currently I prefer to keep the "VBox Guest Additions" versions provided officially by Ubuntu and keep the images smaller.
> 
> If there will be some "killer feature" in the future/latest versions of "VBox Guest Additions" then we can discuss it again...
> 
> Thank you for your help :-)

I asked Hashicorp about this and they told me their base boxes are based on Chef Bento ones. I took a look, and looks like they do the installation from the ISO indeed. 

https://github.com/chef/bento/blob/master/packer_templates/ubuntu/ubuntu-18.04-amd64.json#L32

**Comment by ruzickap** [comment]
I checked it again and you are right. There are several ways how to install VBox Guest Additions.

I tried to install it several ways but the final Ubuntu image was still about 10% bigger than the current one.

I prefer to use the original "old" VBox Guest Additions provided by Ubuntu, because the installation is smaller and doesn't bring any additional complexity to the installation.

I'm not sure what brings "new" VBox Guest Additions to justify 10% bigger boxes (images) - I hope older (but still Ubuntu supported) VBox Guest Additions should be working fine for all users...

(Next LTS Ubuntu version 20.04 should already have VBox Guest Additions in version 6.1.2 (https://packages.ubuntu.com/focal/virtualbox).)

Anyway - let me know if there is some "cool" feature in latest version of "VBox Guest Additions" which you are missing and which is not in "older" version.
-> Then I'll look at it again

**Comment by felipecassiors** [comment]
Great to know about this in Ubuntu 20.04. 

I know that VirtualBox 6.1 renamed the `modifyvm` option `--clipboard` to `--clipboard-mode`, I had to [change this in my `Vagrantfile`](https://github.com/felipecassiors/ubuntu1804-4dev/commit/3a7507bca6b8675db090b17e25db12c262147783#diff-23b6f443c01ea2efcb4f36eedfea9089L17) but didn't test how the old Guest Additions behaves.
